src/pip/_internal/commands/show.py

```diff
@@ -177,4 +198,54 @@ def print_results(distributions, list_files=False, verbose=False):
                 write_output("  %s", line.strip())
             if "files" not in dist:
                 write_output("Cannot locate installed-files.txt")
+
+    return results_printed
+
+
+def print_results_json(distributions, list_files=False, verbose=False):
+    """
+    Build a dictionary with  information from installed distributions found.
+    """
+
+    results_printed = False
+    pkg_infos_list = []
+
+    for i, dist in enumerate(distributions):
+        results_printed = True
+        pkg_info = {}
+
+        pkg_info["Name"] = dist.get('name', '')
+        pkg_info["Version"] = dist.get('version', '')
+        pkg_info["Summary"] = dist.get('summary', '')
+        pkg_info["Home-page"] = dist.get('home-page', '')
+        pkg_info["Author"] = dist.get('author', '')
+        pkg_info["Author-email"] = dist.get('author-email', '')
+        pkg_info["License"] = dist.get('license', '')
+        pkg_info["Location"] = dist.get('location', '')
+        pkg_info["Requires"] = dist.get('requires', [])
+        pkg_info["Required-by"] = dist.get('required_by', [])
+
+        if verbose:
+            pkg_info["Metadata-Version"] = dist.get('metadata-version', '')
+            pkg_info["Installer"] = dist.get('installer', '')
+
+            pkg_info["Classifiers"] = []
+            for classifier in dist.get('classifiers', []):
+                pkg_info["Classifiers"].append(classifier)
+
+            pkg_info["Entry-points"] = []
+            for entry in dist.get('entry_points', []):
+                pkg_info["Entry-points"].append(entry.strip())
+
+        if list_files:
+            if "files" not in dist:
+                pkg_info["Files"] = 'Cannot locate installed-files.txt'
```

Comment:
I'd suggest `None` as value.

Reply:
Hi @xavfernandez 

I will change this to None. But then this will be inconsistent with how the default format does it, will that be a problem?